tests/schedulers/test_schedulers.py

```diff
@@ -309,10 +323,11 @@ def model(sample, t, *args):
 
         return model
 
-    def check_over_configs(self, time_step=0, **config):
+    def check_over_configs(self, time_step=None, **config):
         kwargs = dict(self.forward_default_kwargs)
 
         num_inference_steps = kwargs.pop("num_inference_steps", None)
+        time_step = time_step if time_step is not None else self.default_timestep
```

Comment:
I don't think I understand: the `check_over_configs()` is usually called from the subclass so we can easily provide any value we want from there? for example https://github.com/huggingface/diffusers/blob/958e17dada02c58eabed798260514e8d74edd842/tests/schedulers/test_scheduler_heun.py#L26 

`time_step` is an argument here, so we can easily pass a value that's not default. Why do we need to make it an property and then overwrite it? 

but maybe I misunderstood something, let me know if I did! Thanks:)

Reply:
Hi @yiyixuxu, the change is mostly aimed toward tests like `test_from_save_pretrained`, `test_step_shape`, and `test_scheduler_outputs_equivalence`, all of which currently hardcode timesteps internally. Changing those hardcoded timesteps to be properties allows the timesteps used in those tests to be changed via overriding the properties (e.g. `self.default_timestep`) in a subclass.

Once e.g. the `default_timestep` property exists, I think it makes sense to then make `check_over_configs` and `check_over_forward` default to using it (although I agree that the change itself isn't very useful for them since they already have a `time_step` argument, as you pointed out).

Reply:
@dg845 
ohh I got it now. I did not like these hardcoded 1 and 2 timesteps too. I think it makes a lot of sense to change them.

1. I think you probably don't have to change the default value for `check_over_configs` and `check_over_forward`. But I don't have strong opinions either way:) 
2. I actually think it is might be a mistake to use default `1` and `0` timesteps for scheduler test to begin with. Even if it was not the case, these numbers certainly do not make any sense anymore. I've noticed that some schedulers were be able to handle non-existing timesteps (e.g. for dpm scheduler, it used to default to the last timestep or something when the timestep is out of its range, I've updated the code so it is no longer the case) so the tests were able to run without an error but it was not the intended behavior and these tests were actually invalid.  I wonder if it makes sense to update the default values - maybe instead of `1` and `0`, we default to`timesteps[0]` and `timesteps[1]`. We would need to update some current tests though. cc @patrickvonplaten @DN6 for their opinions before we commit anything

Reply:
Agree with your reasoning @yiyixuxu